SU2_CFD/include/gradients/computeGradientsGreenGauss.hpp

```diff
@@ -153,29 +155,205 @@ void computeGradientsGreenGauss(CSolver* solver,
         if (!nodes->GetDomain(iPoint)) continue;
 
         su2double volume = nodes->GetVolume(iPoint) + nodes->GetPeriodicVolume(iPoint);
-
         const auto area = geometry.vertex[iMarker][iVertex]->GetNormal();
 
         for (size_t iVar = varBegin; iVar < varEnd; iVar++)
-        {
-          su2double flux = field(iPoint,iVar) / volume;
-
-          for (size_t iDim = 0; iDim < nDim; iDim++)
-            gradient(iPoint, iVar, iDim) -= flux * area[iDim];
-        }
-      }
+          flux[iVar] = field(iPoint,iVar) / volume;
+
+        for (size_t iVar = varBegin; iVar < varEnd; iVar++) {
+          for (size_t iDim = 0; iDim < nDim; iDim++) {
+            gradient(iPoint, iVar, iDim) -= flux[iVar] * area[iDim];
+          }
+        } // loop over variables
+      } // vertices
       END_SU2_OMP_FOR
+    } //found right marker
+  } // iMarkers
+
+  /* For symmetry planes, we need to impose the conditions (Blazek eq. 8.40):
+   * 1. n.grad(phi) = 0
+   * 2. n.grad(v.t) = 0
+   * 3. t.grad(v.n) = 0
+   */
+
+  /*--- Check how many symmetry planes there are ---*/
+  unsigned short Syms[MAXNSYMS] = {0};
+  unsigned short nSym = 0;
+  for (size_t iMarker = 0; iMarker < geometry.GetnMarker(); ++iMarker) {
+    if ((config.GetMarker_All_KindBC(iMarker) == SYMMETRY_PLANE) ||
+       (config.GetMarker_All_KindBC(iMarker) == EULER_WALL)) {
+    Syms[nSym] = iMarker;
+    nSym++;
     }
   }
 
+  for (size_t iMarker = 0; iMarker < geometry.GetnMarker(); ++iMarker) {
+
+    if ((config.GetMarker_All_KindBC(iMarker) == SYMMETRY_PLANE) || (config.GetMarker_All_KindBC(iMarker) == EULER_WALL)) {
+      SU2_OMP_FOR_STAT(32)
+      for (size_t iVertex = 0; iVertex < geometry.GetnVertex(iMarker); ++iVertex) {
+
+        size_t iPoint = geometry.vertex[iMarker][iVertex]->GetNode();
+
+        /*--- Normal vector for this vertex (negate for outward convention). ---*/
+        const su2double* VertexNormal = geometry.vertex[iMarker][iVertex]->GetNormal();
+
+        const auto NormArea = GeometryToolbox::Norm(nDim, VertexNormal);
+
+        su2double UnitNormal[nDim] = {0.0};
+        for (size_t iDim = 0; iDim < nDim; iDim++)
+          UnitNormal[iDim] = VertexNormal[iDim] / NormArea;
+
+        /*--- Normal of the primary symmetry plane ---*/
+        su2double NormalPrim[MAXNDIM] = {0.0}, UnitNormalPrim[MAXNDIM] = {0.0};
+
+        /*--- At this point we can find out if the node is shared with another symmetry.
+         * Step 1: do we have other symmetries? ---*/
+        if (nSym>1) {
+          /*--- Step 2: are we on a shared node? ---*/
+          for (auto jMarker=0;jMarker<nSym;jMarker++) {
```

Comment:
This is duplicated in FVMFlowSolverBase, I think this can be a function here, callable from CFVMFlowSolverBase?